addons/html_builder/static/src/website_sale/products_item_option_plugin.js

```diff
@@ -0,0 +1,430 @@
+            position: "left",
+          });
+          this.ribbons.push(ribbon);
+          this.ribbonsObject[ribbonId] = ribbon;
+          return this._setRibbon(editingElement, ribbon);
+        },
+      },
+      modifyRibbon: {
+        getValue: ({ editingElement, param }) => {
+          const field = param.mainParam;
+          let ribbonId = parseInt(editingElement.dataset.ribbonId);
+          if (!ribbonId) return;
+
+          return this.ribbonsObject[ribbonId][field];
+        },
+        isApplied: ({ editingElement, param, value }) => {
+          const field = param.mainParam;
+          let ribbonId = parseInt(editingElement.dataset.ribbonId);
+          if (!ribbonId) return;
+          if (!this.ribbonsObject[ribbonId]) {
+            ribbonId = Object.keys(this.ribbonsObject).find(
+              (key) => this.ribbonsObject[key].id === ribbonId
+            );
+            editingElement.dataset.ribbonId = ribbonId;
+          }
+          return this.ribbonsObject[ribbonId][field] === value;
+        },
+        apply: ({ editingElement, param, value }) => {
+          const isPreviewMode = historyPlugin.getIsPreviewing();
+          const setting = param.mainParam;
+          const ribbonId = parseInt(editingElement.dataset.ribbonId);
+          this.ribbonsObject[ribbonId][setting] = value;
+
+          const ribbon = this.ribbons.find((ribbon) => ribbon.id == ribbonId);
+          ribbon[setting] = value;
+
+          return this._setRibbon(editingElement, ribbon, !isPreviewMode);
+        },
+      },
+      deleteRibbon: {
+        apply: async ({ editingElement }) => {
+          const save = await new Promise((resolve) => {
+            this.services.dialog.add(ConfirmationDialog, {
+              body: _t("Are you sure you want to delete this ribbon?"),
+              confirm: () => resolve(true),
+              cancel: () => resolve(false),
+            });
+          });
+          if (!save) {
+            return;
+          }
+          return this._deleteRibbon(editingElement);
+        },
+      },
+    };
+  }
+
+  async loadRibbons() {
+    return (
+      this.ribbons ||
+      reactive(
+        await this.services.orm.searchRead(
+          "product.ribbon",
+          [],
+          ["id", "name", "bg_color", "text_color", "position"]
+        )
+      )
+    );
+  }
+
+  async getDefaultSort() {
+    return (
+      this.defaultSort ||
+      (await this.services.orm.searchRead(
+        "website",
+        [["id", "=", this.currentWebsiteId]],
+        ["shop_default_sort"]
+      ))
+    );
+  }
+
+  _setRibbon(editingElement, ribbon, save = true) {
+    const ribbonId = ribbon.id;
+    const editableDocument = editingElement.ownerDocument.body;
+    editingElement.dataset.ribbonId = ribbonId;
+
+    // Find or create ribbon element
+    let ribbonElement = editingElement.querySelector(".o_ribbon");
+    if (!ribbonElement && ribbonId) {
+      ribbonElement = document.createElement("span");
```

Comment:
just noticed that you used the global document. This is not the same document as the document inside the iframe. Plugins have direct access to `this.document` which is very likely what you want to use here